### PR TITLE
solve the problem of safe calculation by removing liquidity

### LINKED ISSUE #333: AmmSwap:Removing liquidity causes zero attacks
## Summary of Bug
When liquidity is deleted, it may cause one token balance in the pool to be 0, while the other is not, which will lead to the abnormal operation of this coin to the pool

### Problem Definition

1. Because okexchain only supports 8 decimal places, the 8*10^-9 is rounded to 10^-8, so the user gets 8 tokenA and 10^-8 okt
2. This results in a token pair pool of tokenA_okt with some tokenA and zero okt left
3. This will affect the numerical calculation for adding liquidity and exchanging tokens

### PR DESCRIPTION
**Pull request by Yoio**
Fixes #333 

- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/okex/okexchain/blob/develop/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/okex/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: x/ammswap/handler.go [hunk 1]
```diff
@@ -452,5 +456,5 @@ var (
 // mulAndQuo returns a * b / c
 func mulAndQuo(a, b, c sdk.Dec) sdk.Dec {
 	a = a.Mul(auxiliaryDec)
-	return a.Mul(b).Quo(c).Quo(auxiliaryDec)
+	return a.Mul(b).Quo(c).QuoTruncate(auxiliaryDec)
```
Comment (xiangjianmeng) [inline x/ammswap/handler.go:459]: it is suggested to change all of mul or quo to MulTruncate or QuoTruncate in ammswap module.

File: x/ammswap/handler.go [hunk 2]
```diff
@@ -295,7 +300,7 @@ func handleMsgTokenToNativeToken(ctx sdk.Context, k Keeper, msg types.MsgTokenTo
 	}
 	params := k.GetParams(ctx)
 	tokenBuy := calculateTokenToBuy(swapTokenPair, msg, params)
-	if tokenBuy.Amount.LT(msg.MinBoughtTokenAmount.Amount) {
+	if tokenBuy.IsZero() || tokenBuy.Amount.LT(msg.MinBoughtTokenAmount.Amount) {
 		return sdk.Result{
 			Code: sdk.CodeInternal,
 			Log:  fmt.Sprintf("Failed: expected minimum token to buy is %s but got %s", msg.MinBoughtTokenAmount, tokenBuy),
```
Comment (xiangjianmeng) [inline x/ammswap/handler.go:306]: It is suggested to return log like "selled token amount is too little to buy any token" when tokenBuy.IsZero() is true.

File: x/ammswap/handler.go [hunk 3]
```diff
@@ -350,12 +355,18 @@ func handleMsgTokenToToken(ctx sdk.Context, k Keeper, msg types.MsgTokenToNative
 	msgOne := msg
 	msgOne.MinBoughtTokenAmount = nativeAmount
 	tokenNative := calculateTokenToBuy(swapTokenPairOne, msgOne, params)
-
+	if tokenNative.IsZero() {
+		return sdk.Result{
+			Code: sdk.CodeInternal,
+			Log:  fmt.Sprintf("Failed: expected minimum token to buy is zero"),
```
Comment (xiangjianmeng) [inline x/ammswap/handler.go:361]: ditto

File: x/ammswap/handler.go [hunk 4]
```diff
@@ -350,12 +355,18 @@ func handleMsgTokenToToken(ctx sdk.Context, k Keeper, msg types.MsgTokenToNative
...
-	tokenBuy := calculateTokenToBuy(swapTokenPairOne, msgTwo, params)
-
-	if tokenBuy.Amount.LT(msg.MinBoughtTokenAmount.Amount) {
+	tokenBuy := calculateTokenToBuy(swapTokenPairTwo, msgTwo, params)
+	// sanity check. user may set MinBoughtTokenAmount to zero on front end.
+	// if set zero,this will not return err
+	if tokenBuy.IsZero() || tokenBuy.Amount.LT(msg.MinBoughtTokenAmount.Amount) {
 		return sdk.Result{
 			Code: sdk.CodeInternal,
 			Log:  fmt.Sprintf("Failed: expected minimum token to buy is %s but got %s", msg.MinBoughtTokenAmount, tokenBuy),
```
Comment (xiangjianmeng) [inline x/ammswap/handler.go:372]: ditto

**Review by xiangjianmeng — APPROVED** [review]

**Review by summerpro — APPROVED** [review]

**Pull request merged**